interfaces/adfsuite/ams.py

```diff
@@ -2193,15 +2237,20 @@ class AMSJob(SingleJob):
     """A class representing a single computation with AMS driver. The corresponding results type is |AMSResults|."""
 
     results: AMSResults
-    molecule: Union[Molecule, Dict[str, Molecule], None]
+    molecule: Optional[Union[Molecule, Dict[str, Molecule], "ChemicalSystem"]]
```

Comment:
Isn't a ``Dict[str, "ChemicalSystem"]`` allowed here too if you want multiple System blocks in the input?

Reply:
Yes good point, updated and added a test for NEB with two system block (also for the dict of molecules case)